plugin/store/store.go

```diff
@@ -98,12 +98,31 @@ func (ps *Store) SetState(p *v2.Plugin, state bool) {
 }
 
 // Add adds a plugin to memory and plugindb.
-func (ps *Store) Add(p *v2.Plugin) {
+// An error will be returned if there is a collision.
```

Comment:
@yongtang : Not this change, but `Pull` should also remove the created pluginDir, in case of errors. Would you like to take care of that as part of this PR?